### PR TITLE
ImageServing performance improvements - phase #2

### PR DESCRIPTION
**Pull request by macbre**
For the first phase see #6232

https://wikia-inc.atlassian.net/browse/PLATFORM-817

This change will make ImageServing use pre-generated reports stored in `querycache` table instead of making selects with union on each request.

Before the change:

``` sql
> explain (SELECT /* ImageServingDriverMainNS::getImagesPopularity::imagelinks 127.0.0.1 */ 'Ulica_Strzelecka_1917.jpg' AS il_to FROM `imagelinks` JOIN `page` on page.page_id = il_from WHERE il_to = 'Ulica_Strzelecka_1917.jpg' AND page_namespace IN (0) LIMIT 11 ) UNION ALL (SELECT 'IMG_7303.jpg' AS il_to FROM `imagelinks` JOIN `page` on page.page_id = il_from WHERE il_to = 'IMG_7303.jpg' AND page_namespace IN (0) LIMIT 11 );
+----+--------------+------------+--------+--------------------+---------+---------+-----------------------------+------+--------------------------+
| id | select_type  | table      | type   | possible_keys      | key     | key_len | ref                         | rows | Extra                    |
+----+--------------+------------+--------+--------------------+---------+---------+-----------------------------+------+--------------------------+
|  1 | PRIMARY      | imagelinks | ref    | il_from,il_to      | il_to   | 257     | const                       |    3 | Using where; Using index |
|  1 | PRIMARY      | page       | eq_ref | PRIMARY,name_title | PRIMARY | 4       | plpoznan.imagelinks.il_from |    1 | Using where              |
|  2 | UNION        | imagelinks | ref    | il_from,il_to      | il_to   | 257     | const                       |    3 | Using where; Using index |
|  2 | UNION        | page       | eq_ref | PRIMARY,name_title | PRIMARY | 4       | plpoznan.imagelinks.il_from |    1 | Using where              |
| NULL | UNION RESULT | <union1,2> | ALL    | NULL               | NULL    | NULL    | NULL                        | NULL |                          |
+----+--------------+------------+--------+--------------------+---------+---------+-----------------------------+------+--------------------------+
5 rows in set (0.00 sec)
```

After the change:

``` sql
> explain SELECT /* ImageServingDriverMainNS::getImagesPopularity 127.0.0.1 */  qc_title as image,qc_value as popularity  FROM `querycache`  WHERE qc_type = 'MostimagesInContent' AND qc_title IN ('Ulica_Strzelecka_1917.jpg','IMG_7303.jpg','Ulica_23_Lutego_zniszczenia_wojenne.jpg','Nazwy_ulic_w_Poznaniu_z_planem_Wielkiego_Poznania.jpg','Koziołki.jpg','Citroen_2CV6_and_Poznań_Cathedral_07-Oct-2011.JPG','Ratusz_ujęcie_od_dołu.jpg','Kamieniczki_budnicze.png','TEY.jpg');
+----+-------------+------------+------+---------------+---------+---------+-------+------+-------------+
| id | select_type | table      | type | possible_keys | key     | key_len | ref   | rows | Extra       |
+----+-------------+------------+------+---------------+---------+---------+-------+------+-------------+
|  1 | SIMPLE      | querycache | ref  | qc_type       | qc_type | 34      | const |    1 | Using where |
+----+-------------+------------+------+---------------+---------+---------+-------+------+-------------+
1 row in set (0.00 sec)
```

@michalroszka / @wladekb / @drozdo

### CODE REVIEW

File: extensions/wikia/ImageServing/drivers/ImageServingDriverMainNS.class.php [hunk 1]
```diff
@@ -160,74 +161,55 @@ protected function loadImageDetails( $imageNames = array() ) {
...
-		global $wgContentNamespaces;
-
+	/**
+	 * Returns the popularity of given set of images (popularity = the number of articles in content namespaces that use an image)
+	 *
+	 * $limit is applied - images that have popularity that is higher than $limit will not be returned
+	 *
+	 * Example:
+	 *
+	 * $imageNames = [ 'IMG_7303.jpg', 'Koziołki.jpg' ]
```
Comment (wladekb) [inline extensions/wikia/ImageServing/drivers/ImageServingDriverMainNS.class.php]: Let's keep the examples in English unless they are language-specific. Imagine yourself seeing a Chinese text in phpdoc ;-)

Comment (macbre) [inline extensions/wikia/ImageServing/drivers/ImageServingDriverMainNS.class.php]: True, changed in  a1cdb27 :)

File: includes/wikia/services/AvatarService.class.php [hunk 2]
```diff
@@ -269,7 +269,7 @@ public static function getVignetteUrl( Masthead $masthead, $width, $timestamp )
 				$url = self::buildVignetteUrl( $width, $bucket, $relativePath, $timestamp, false );
 			}
 		} else { // default avatar
-			$legacyDefaultUrl = array_shift( $masthead->getDefaultAvatars( 'thumb/' ) );
+			$legacyDefaultUrl = @array_shift( $masthead->getDefaultAvatars( 'thumb/' ) );
```
Comment (wladekb) [inline includes/wikia/services/AvatarService.class.php]: Using @ is deemed to be a bad practice. Can we handle this in some other way or at least add a comment saying explicitly what is the case we want to hide warnings for?

**Comment by wladekb** [comment]
awesome (=lgtm)

File: includes/wikia/services/AvatarService.class.php (hunk 2 above)
Comment (macbre) [inline includes/wikia/services/AvatarService.class.php]: Right, that was a quick fix as strict warnings were driving me crazy :)

Sure, needs to be properly fixed. Storing the function result in a temporary variable that will be passed to `array_shift`.

Comment (macbre) [inline includes/wikia/services/AvatarService.class.php]: Temp fix reverted in  b85ab7e

**Pull request merged**